worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -193,6 +193,68 @@ function applyActionDetails(document, actionDetails, terminateOnFailure) {
     }
 }
 
+function onAfterProcessDocument(e) {
+    if (!e.application.getInputMessageProcessor().getProcessSubdocumentsSeparately() && e.rootDocument.hasSubdocuments()) {
+        processSubdocumentFailures(e.rootDocument.getSubdocuments());
+    }
+    processFailures(e.document);
+}
+
+function processSubdocumentFailures(subdocuments) {
+    for each(var subdoc in subdocuments) {
+        if (subdoc.hasSubdocuments()) {
+            processSubdocumentFailures(subdoc.getSubdocuments());
+        } else {
+            processFailures(subdoc);
+        }
+    }
+}
+
+function processFailures(document) {
+    if (document.getFailures().isChanged()) {
+
+        var listOfFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfFailures.add(failure);
+        });
+
+        document.getFailures().reset();
+
+        var listOfOriginalFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfOriginalFailures.add(failure);
+        });
+
+        for each (var f in listOfFailures) {
+            if (!isFailureInOriginal(listOfOriginalFailures, f)) {
+                var message = {
+                    id: f.getFailureId(),
+                    stack: f.getFailureStack() || undefined,
+                    source: document.rootDocument.getField("CAF_WORKFLOW_ACTION").getStringValues().get(0),
+                    version: document.getTask().getService(com.hpe.caf.api.worker.WorkerTaskData.class).getSourceInfo().getVersion(),
+                    workflowName: document.getField("CAF_WORKFLOW_NAME").getStringValues().get(0),
+                    originalDescription: f.getFailureMessage()
+                };
```

Comment:
- I'd like the time added to the structure, so that we have the failure times.
- I think we may need to change the field names from camelCase to MACRO_CASE.

Reply:
done